### ISSUE TITLE
XmlRecordReader return a ClassCastException when reading a comment line

### ISSUE THREAD

**Issue by vian42** [j-easy/easy-batch#337]
Hi, on the project we actually use easy-batch 5.2 with the JDK 11.

if the input xml file have a comment line (with < ! --   -->), this method "xmlEventReader.nextEvent()" return a com.sun.xml.internal.stream.events.**CommentEvent** (org.easybatch.xml.XmlRecordReader.readRecord() line 91)

when this occurred the method "xmlEvent.asCharacters()" (same method as before, line 97) throw the ClassCastException.

Shouldn't we call the method "xmlEvent.**is**Characters()" before, and ignore the comment from the xml file ? 
Or does exist a way to manage those comments?

<img width="551" alt="2019_04_03_15_46_43_" src="https://user-images.githubusercontent.com/48093456/55485885-9f6fd680-562b-11e9-9256-a84ffdc26cc0.png">
<img width="276" alt="2019_04_03_15_51_02_Evaluate" src="https://user-images.githubusercontent.com/48093456/55485889-a0a10380-562b-11e9-892e-a181c626b45f.png">

**Comment by benas** [comment]
Good catch! Looks like a bug in easy batch.

> Shouldn't we call the method "xmlEvent.isCharacters()" before, and ignore the comment from the xml file ?

Good idea! Do you want to contribute back with a PR? You are welcome.

**Comment by vian42** [comment]
So the idea is to replace the else by: 
```java
                } else if (xmlEvent.isCharacters()) {
                    stringBuilder.append(escape(xmlEvent.asCharacters().getData()));
                } else {
                    LOGGER.info("The line was not parse : " + xmlEvent.toString());
                }
```
but I not sure what to put in the last else. Is the log a good idea?

**Comment by dpasdar** [comment]
I have arrived at the same conclusion/solution, I would like to contribute with a PR, however I can't create branches in this project.

**Comment by laxika** [comment]
@dpasdar Fork it and create a branch in your fork. You will be able to create a PR from there.

**Comment by benas** [comment]
@vian42 Thank you for reporting this. It is indeed a bug in Easy Batch. It it fixed in f68ed2a.

> not sure what to put in the last else. Is the log a good idea?

No need for the log, comments can be silently ignored.
